src/main/webviews/detailsWebView.ts

```diff
@@ -10,39 +10,41 @@ import { LogManager } from '../log/logManager';
 export class DetailsWebView {
     constructor(private _logManager: LogManager) {}
 
-    public async activate(context: vscode.ExtensionContext) {
-        let panel: vscode.WebviewPanel | undefined;
-        let prevActiveTreeNode: IDependencyPage;
+    private _panel: vscode.WebviewPanel | undefined;
+    private _currentData: any;
 
+    public async activate(context: vscode.ExtensionContext) {
         context.subscriptions.push(
-            vscode.commands.registerCommand('jfrog.view.dependency.details.page', (page: IDependencyPage) => {
-                if (!panel) {
-                    panel = createWebview(context);
-                    panel.onDidChangeViewState(e => {
-                        updateWebview(e.webviewPanel, prevActiveTreeNode);
-                    });
-                    panel.onDidDispose(
-                        () => {
-                            panel = undefined;
-                        },
-                        undefined,
-                        context.subscriptions
-                    );
-                } else {
-                    panel.reveal();
-                }
-
-                if (page && prevActiveTreeNode !== page) {
-                    this._logManager.logMessage('Opening webview page with data:\n' + JSON.stringify(page), 'DEBUG');
-                    updateWebview(panel, page);
-                }
-                prevActiveTreeNode = page;
-            })
+            vscode.commands.registerCommand('jfrog.view.details.page', (page: IDependencyPage | IEosPage) => this.updateWebview(page, context))
         );
     }
+
+    public updateWebview(data: any, context: vscode.ExtensionContext) {
+        if (!this._panel) {
+            this._panel = createWebview(context);
+            this._panel.onDidChangeViewState(e => {
```

Comment:
`e` is not enough informative
```suggestion
            this._panel.onDidChangeViewState((webviewChanged: vscode.WebviewPanelOnDidChangeViewStateEvent) => {
```